conans/cli/api/subapi/install.py

```diff
@@ -38,9 +42,64 @@ def install_consumer(deps_graph, generators=None, source_folder=None, output_fol
         conanfile.folders.set_base_generators(output_folder)
         conanfile.folders.set_base_build(output_folder)
 
+        _do_deploys(self.conan_api, conanfile, deploy, output_folder)
+
         # Add cli -g generators
         conanfile.generators = list(set(conanfile.generators).union(generators or []))
         write_generators(conanfile)
 
         if type(conanfile).system_requirements != ConanFile.system_requirements:
             call_system_requirements(conanfile)
+
+
+def _do_deploys(conan_api, conanfile, deploy, output_folder):
+    # Handle the deploys
+    cwd = os.getcwd()
+    for d in deploy or []:
+        # First, find the file containing the deployer, could be local or in the cache
+        if d.startswith("conan"):  # built-in!
+            deployer = {"conan_full_deploy": conan_full_deploy}[d]
+        else:
+            if not d.endswith(".py"):
+                d += ".py"  # Deployers must be python files
+            if not os.path.isabs(d):
+                full_path = os.path.normpath(os.path.join(cwd, d))  # Try first local path
+                if not os.path.isfile(full_path):  # Then try in the cache
+                    full_path = os.path.join(conan_api.cache_folder, "extensions", "deploy", d)
+                    if not os.path.isfile(full_path):
+                        raise ConanException(f"Cannot find deployer '{d}'")
+            else:
+                full_path = d
+                if not os.path.isfile(full_path):
+                    raise ConanException(f"Cannot find deployer '{d}'")
+
+            mod, _ = load_python_file(full_path)
+            deployer = mod.deploy
+
+        # IMPORTANT: Use always kwargs to not break if it changes in the future
+        deployer(conanfile=conanfile, output_folder=output_folder)
+
+
+def conan_full_deploy(conanfile, output_folder):
+    """
+    Deploys to output_folder + host/dep/0.1/Release/x86_64 subfolder
+    """
+    # TODO: Maybe we should receive the full graph instead? Receive the full conan_api too?
+    # TODO: This deployer needs to be put somewhere else
+    import os
+    import shutil
+
+    conanfile.output.info(f"Conan built-in full deployer to {output_folder}")
+    for r, d in conanfile.dependencies.items():
+        folder_name = os.path.join(d.context, d.ref.name, str(d.ref.version))
+        build_type = str(d.info.settings.build_type)
+        arch = str(d.info.settings.arch)
+        if build_type:
+            folder_name = os.path.join(folder_name, build_type)
+        if arch:
+            folder_name = os.path.join(folder_name, arch)
+        new_folder = os.path.join(output_folder, folder_name)
+        # TODO: This deployer doesn't work with editable (and it would be difficult)
+        #  we could raise an error if editable, but not easy to know it, not in ConanFileInterface
+        shutil.copytree(d.package_folder, new_folder)
```

Comment:
Running the deploy twice, over the same location is going to fail because the folders already exist

Reply:
Yes, good point. And probably that is what we want unless you pass a ``--force`` arg? We have to define the UX around this use case, indeed.

Reply:
Yes, I think we should decide what Conan should do in these cases and make it behave coherently for all commands...

Reply:
It is a bit variable:

- For files, like the ones generated by generators like CMakeDeps, they simply overwrite each time, and it is necessary they do, otherwise in practice it would be really bad UX, and --force would be mandatory
- The ``conan new`` command doesn't overwrite, unles ``--force``, because it is likely that user did local changes to the created files that would be lost otherwise. And it doesn't seem a common use case to be calling ``conan new`` repeteadly
- ``conan profile detect`` also will fail if the file already exist, for the same reason: it could be losing user changes

For this case, as all files are just a copy from the conan cache, I'd say overwriting by default should be ok, but lets discuss it.

Reply:
Yes, for this case it should probably behave like the generators. In case you don't want to behave that way the deploy you can always implement other logic in your custom deployer...